packages/expo-router/src/link/zoom/useZoomTransitionPrimitives.ios.tsx

```diff
@@ -5,19 +5,40 @@ import { useCallback, useEffect, useMemo, useState } from 'react';
 
 import { isZoomTransitionEnabled } from './ZoomTransitionEnabler';
 import { INTERNAL_EXPO_ROUTER_ZOOM_TRANSITION_SOURCE_ID_PARAM_NAME } from '../../navigationParams';
-import { parseUrlUsingCustomBase } from '../../utils/url';
+import { parseUrlUsingCustomBase, shouldLinkExternally } from '../../utils/url';
 import { useIsPreview } from '../preview/PreviewRouteContext';
 import { LinkProps } from '../useLinkHooks';
+import type { ZoomTransitionSourceContextValueType } from './zoom-transition-context';
 
 export function useZoomTransitionPrimitives({ href, asChild }: LinkProps) {
   const isPreview = useIsPreview();
-  const zoomTransitionId = useMemo(
-    () =>
-      !isPreview && process.env.EXPO_OS === 'ios' && isZoomTransitionEnabled()
-        ? nanoid()
-        : undefined,
-    []
-  );
+  const isExternalHref =
+    typeof href === 'string' ? shouldLinkExternally(href) : shouldLinkExternally(href.pathname);
+
+  const isZoomTransitionConfigValid =
+    process.env.EXPO_OS === 'ios' &&
+    !isPreview &&
+    !!isZoomTransitionEnabled() &&
+    !!asChild &&
+    !isExternalHref;
+
+  const reason = useMemo(() => {
+    if (process.env.EXPO_OS !== 'ios' || isPreview) {
+      return undefined;
+    }
+    if (!isZoomTransitionEnabled()) {
+      return 'Zoom transitions are not enabled.';
+    }
+    if (!asChild) {
+      return 'Link must be used with `asChild` prop to enable zoom transitions.';
+    }
+    if (isExternalHref) {
+      return 'Zoom transitions can only be used with internal links.';
+    }
+    return undefined;
+  }, [asChild, isExternalHref]);
+
+  const zoomTransitionId = useMemo(nanoid, []);
```

Comment:
Didn't spot this in a prior PR, I think. Could this be using `useId` instead? `nanoid` seems aggressive here, and `useId` seems like it could be a good fit https://react.dev/reference/react/useId

Alternatively, since this is SSR-less and will always be SSR-less, I'm assuming this is a downstream requirement. Would we be able to use a counter here?